packages/css/src/components/radio/radio.scss

```diff
@@ -44,111 +23,181 @@
   gap: var(--ams-radio-gap);
   line-height: var(--ams-radio-line-height);
   outline-offset: var(--ams-radio-outline-offset);
+  position: relative;
+  text-decoration-thickness: var(--ams-radio-text-decoration-thickness);
+  text-underline-offset: var(--ams-radio-text-underline-offset);
+
+  @include text-rendering;
+}
 
-  &:hover {
-    color: var(--ams-radio-hover-color);
-    text-decoration-line: underline;
-    text-decoration-thickness: var(--ams-radio-hover-text-decoration-thickness);
-    text-underline-offset: 0.375rem;
+.ams-radio__icon-container {
+  block-size: var(--ams-radio-icon-container-block-size);
+  display: flex;
+  flex-shrink: 0;
+  inline-size: var(--ams-radio-icon-container-inline-size);
+}
 
-    .ams-radio__circle::after {
-      border-color: var(--ams-radio-circle-hover-border-color);
-    }
+.ams-radio__icon {
+  circle:first-child {
+    stroke: var(--ams-radio-icon-outer-stroke);
+    stroke-width: 0.15rem;
   }
 
-  @include text-rendering;
+  circle:nth-child(2) {
```

Comment:
Radio.svg is pretty much static, so you could've also used BEM to classes inside the SVG. For example `ams-radio__circle-inner` or something. Then if another party wants to customize radio for whatever reason they are not bound by the number of children in the SVG but can give classes to the components of the radio icon.

Reply:
Even more so: if we write CSS for icons, their SVG markup has effectively become API. We need semantic names for the circle and the dot, and use them in SVG, CSS, and tokens.

Reply:
Good point. I originally tried this approach, but SVGR changes the class names when we convert from SVG to React component, so the final class names were a bit unpredictable. And I was too lazy to look into our SVGR config 😁 

I'll look into it now, I agree it's the better approach